test_pylint.py has lots of failures because qtawesome says 'Invalid font prefix "mdi"'

<!--- **PLEASE READ:** When submitting here, please ensure you've completed the following checklist and checked the boxes to confirm. Issue reports without it may be closed. Thanks! --->

### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

This is the last of the test failure reports, and I'm sorry to end on such a weird one.  `spyder/plugins/pylint/tests/test_pylint.py` has lots of failing tests because it seems that it doesn't load or initialise qtawesome correctly.  It's strange, because the Spyder application itself appears to have all of the icons in place.

### What steps reproduce the problem?

(sid):~/debian/spyder-packages/spyder/build-area/spyder-5.2.0+dfsg1 $ CI=true xvfb-run -a -s "-screen 0 1024x768x24 +extension GLX" python3.9 -m pytest spyder/plugins/pylint/tests/test_pylint.py
============================= test session starts ==============================
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
PyQt5 5.15.6 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /home/jdg/debian/spyder-packages/spyder/build-area/spyder-5.2.0+dfsg1, configfile: pytest.ini
plugins: typeguard-2.11.1, flaky-3.7.0, xvfb-2.0.0, mock-3.6.1, cov-3.0.0, order-1.0.0, qt-4.0.2, lazy-fixture-0.6.3
collected 22 items                                                             

spyder/plugins/pylint/tests/test_pylint.py ..........EEEEEEEEEEEE        [100%]

==================================== ERRORS ====================================
________________ ERROR at setup of test_pylint_widget_noproject ________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x7fe595ffc9a0>
qtbot = <pytestqt.qtbot.QtBot object at 0x7fe595ffcac0>

    @pytest.fixture
    def pylint_plugin(mocker, qtbot):
        main_window = MainWindowMock()
        main_window.projects.get_active_project_path = mocker.MagicMock(
            return_value=None)
>       plugin = Pylint(parent=main_window, configuration=CONF)

spyder/plugins/pylint/tests/test_pylint.py:79: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
spyder/api/plugins/new_api.py:915: in __init__
    super().__init__(parent, configuration=configuration)
spyder/api/plugins/new_api.py:313: in __init__
    self._container = container = self.CONTAINER_CLASS(
spyder/plugins/pylint/main_widget.py:297: in __init__
    super().__init__(name, plugin, parent)
spyder/api/widgets/main_widget.py:244: in __init__
    self._main_toolbar = MainWidgetToolbar(
spyder/api/widgets/toolbars.py:298: in __init__
    super().__init__(parent, title=title or '')
spyder/api/widgets/toolbars.py:122: in __init__
    ext_button.setIcon(ima.icon('toolbar_ext_button'))
spyder/utils/icon_manager.py:382: in icon
    return qta.icon(*args, **kwargs)
/usr/lib/python3/dist-packages/qtawesome/__init__.py:188: in icon
    return _instance().icon(*names, **kwargs)
/usr/lib/python3/dist-packages/qtawesome/iconic_font.py:345: in icon
    parsed_options.append(self._parse_options(specific_options,
/usr/lib/python3/dist-packages/qtawesome/iconic_font.py:397: in _parse_options
    prefix, chars = self._get_prefix_chars(names)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <qtawesome.iconic_font.IconicFont object at 0x7fe596b6f0d0>
names = ['mdi.dots-horizontal', 'mdi.dots-horizontal', 'mdi.dots-horizontal', 'mdi.dots-horizontal', 'mdi.dots-horizontal', 'mdi.dots-horizontal', ...]

    def _get_prefix_chars(self, names):
        chars = []
        for name in names:
            if '.' in name:
                prefix, n = name.split('.')
                if prefix in self.charmap:
                    if n in self.charmap[prefix]:
                        chars.append(self.charmap[prefix][n])
                    else:
                        error = 'Invalid icon name "{0}" in font "{1}"'.format(
                            n, prefix)
                        raise Exception(error)
                else:
                    error = 'Invalid font prefix "{0}"'.format(prefix)
>                   raise Exception(error)
E                   Exception: Invalid font prefix "mdi"

/usr/lib/python3/dist-packages/qtawesome/iconic_font.py:431: Exception
[... lots of similar cases skipped ...]
=============================== warnings summary ===============================
../../../../../../../usr/lib/python3/dist-packages/qtawesome/iconic_font.py:354
  /usr/lib/python3/dist-packages/qtawesome/iconic_font.py:354: UserWarning: You need to have a running QApplication to use QtAwesome!
    warnings.warn("You need to have a running "

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_noproject
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[None]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Script]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Working]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Project]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Home]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Script & Home]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Working & Project]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Script & Working]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[Project & Home]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_widget_pylintrc[All]
ERROR spyder/plugins/pylint/tests/test_pylint.py::test_pylint_max_history_conf
=================== 10 passed, 1 warning, 12 errors in 2.20s ===================
```
The same behaviour occurs inside a chroot and in a normal environment, with Python 3.9 and 3.10, and with `xvfb` and without.  I've explored a bit but am none the wiser...

## Versions
<!--- You can get this information from Help > About Spyder...
or (if Spyder won't launch) the "conda list" command
from the Anaconda Prompt/Terminal/command line. --->

* Spyder version: 5.2.0
* Python version: 3.9 and 3.10
* Qt version: 5.15.2
* PyQt version: 5.15.6+dfsg-1+b1
* Operating System name/version: Debian testing and unstable

### Dependencies
<!--- Please go to the menu entry Help > Dependencies,
press the Copy to clipboard button and paste below --->

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0               :  7.27.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jellyfish >=0.7               :  0.8.9 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.3.0 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.1 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.7.1 (OK)
pylint >=2.5.0                :  2.12.2 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.3.1;<1.4.0          :  1.3.2 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.1 (OK)
qtconsole >=5.2.1;<5.3.0      :  5.2.1 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  58.2.0 (OK)
sphinx >=0.6.6                :  4.3.1 (OK)
spyder_kernels >=2.2.0;<2.3.0 :  2.2.0 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.27 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  0.29.24 (OK)
matplotlib >=2.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.21.4 (OK)
pandas >=1.1.1                :  1.1.5 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  1.7.1 (OK)
```

--- a/spyder/plugins/pylint/tests/test_pylint.py
+++ b/spyder/plugins/pylint/tests/test_pylint.py
@@ -15,7 +15,15 @@ from unittest.mock import Mock, MagicMock
 # Third party imports
 import pytest
 from qtpy.QtCore import Signal
-from qtpy.QtWidgets import QMainWindow
+from qtpy.QtWidgets import QApplication, QMainWindow
+
+# This is necessary to run these tests independently from the rest in our
+# test suite.
+# NOTE: Don't move it to another place; it needs to be before importing the
+# Pylint plugin below.
+# Fixes spyder-ide/spyder#17071
+if QApplication.instance() is None:
+    app = QApplication([])
 
 # Local imports
 from spyder.api.plugin_registration.registry import PLUGIN_REGISTRY

--- a/spyder/plugins/pylint/tests/test_pylint.py
+++ b/spyder/plugins/pylint/tests/test_pylint.py
@@ -60,12 +60,11 @@ bad-names={bad_names}
 good-names=e
 """
 
-
 class MainWindowMock(QMainWindow):
     sig_editor_focus_changed = Signal(str)
 
     def __init__(self):
-        super(MainWindowMock, self).__init__(None)
+        super().__init__(None)
         self.editor = Mock()
         self.editor.sig_editor_focus_changed = self.sig_editor_focus_changed
         self.projects = MagicMock()
@@ -82,15 +81,22 @@ class MainWindowMock(QMainWindow):
 @pytest.fixture
 def pylint_plugin(mocker, qtbot):
     main_window = MainWindowMock()
+    main_window.resize(640, 480)
     main_window.projects.get_active_project_path = mocker.MagicMock(
         return_value=None)
+    main_window.show()
+
     plugin = Pylint(parent=main_window, configuration=CONF)
     plugin._register()
     plugin.set_conf("history_filenames", [])
+
     widget = plugin.get_widget()
+    widget.resize(640, 480)
     widget.filecombo.clear()
-    qtbot.addWidget(widget)
-    yield plugin
+    widget.show()
+
+    qtbot.addWidget(main_window)
+    return plugin
 
 
 @pytest.fixture

